### PR TITLE
Solution react_movies-list-add-form

### PR DESCRIPTION
**Pull request by NazariiAlieksieiev**
[DEMO LINK](https://NazariiAlieksieiev.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,45 +1,99 @@
...
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```
Comment (anstsot) [inline src/components/NewMovie/NewMovie.tsx:14]: please create one state `newMovie` as `Movie` type without id and set all inputs there.

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,99 @@
...
+
+  const isValid = (url: string): boolean => {
+    const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/; // eslint-disable-line max-len
+
+    return pattern.test(url);
+  };
+
+  const isValidUrls = isValid(imgUrl) && isValid(imdbUrl);
+
+  const isFormValid = title && isValidUrls && imdbId;
```
Comment (anstsot) [inline src/components/NewMovie/NewMovie.tsx:24]: please check the empty fields because it's possible to add a movie with spaces only.
image

**Review by anstsot — CHANGES_REQUESTED** [review]
It’s almost done! Let’s improve a little

**Review by anstsot — CHANGES_REQUESTED** [review]
It's still possible to add movie with empty fields. Try to use `trim()` to cut the spaces.
image

**Review by IvanFesenko — APPROVED** [review]

**Pull request closed without merging**